packages/web-components/fast-components/src/select/select.styles.ts

```diff
@@ -29,26 +32,44 @@ import {
     neutralFillInputRest,
     neutralFillStealthRest,
     neutralForegroundRest,
-    neutralLayerFloating,
-    neutralStrokeRest,
     strokeWidth,
     typeRampBaseFontSize,
     typeRampBaseLineHeight,
 } from "../design-tokens";
+import { optionHeight } from "../listbox-option/listbox-option.styles";
+import { listboxStrokeOffset, listboxStyles } from "../listbox/listbox.styles";
 import { elevation } from "../styles/elevation";
 import { heightNumber } from "../styles/size";
 
 /**
- * Styles for Select
+ * The width of the widest option.
+ *
+ * @public
+ */
+export const selectOptionWidth = DesignToken.create<number>("option-width");
+
+export const selectMaxSize = DesignToken.create<number>("select-max-height");
+
+/**
+ * Styles for Select.
+ *
  * @public
  */
 export const selectStyles: FoundationElementTemplate<ElementStyles, SelectOptions> = (
     context,
     definition
-) =>
-    css`
-    ${display("inline-flex")} :host {
+) => {
+    const selectContext = context.name === context.tagFor(Select);
+
+    // The expression interpolations present in this block cause Prettier to generate
+    // various formatting bugs.
+    // prettier-ignore
+    return css`
+    ${display("inline-flex")}
+
+    :host {
         --elevation: 14;
+        --indicator-width: calc(${designUnit} * 4px);
```

Comment:
Ideally we don't add a fixed setting here either, and base it on padding. This is roughly equivalent to a hard-coded "16px".

Reply:
I see you've just moved this code. Having it here implies someone might change the css prop. We should probably refactor this with the density work, but I want to make sure we don't publish any new control points we will soon want to remove.

Reply:
I find local CSS custom properties to be super useful, especially for repetitive partials like this. For consistency and to avoid points of confusion, I removed the local variable and reset the values to their original states.